File: app/src/main/java/ai/elimu/herufi/BaseApplication.java
package ai.elimu.herufi;

import android.app.Application;
import android.speech.tts.TextToSpeech;
import android.util.Log;

public class BaseApplication extends Application {

    private TextToSpeech tts;

    @Override
    public void onCreate() {
        Log.i(getClass().getName(), "onCreate");
        super.onCreate();

        // Initialize Text-to-Speech
        getTTS();
    }

    public TextToSpeech getTTS() {
        if (tts == null) {
            tts = new TextToSpeech(getApplicationContext(), new TextToSpeech.OnInitListener() {
                @Override
                public void onInit(int status) {
                    Log.i(getClass().getName(), "TextToSpeech onInit");
                }
            });
        }
        return tts;
    }

    @Override
    public void onTerminate() {
        Log.i(getClass().getName(), "onTerminate");
        super.onTerminate();

        // Release the resources used by the TextToSpeech engine
        if (tts != null) {
            tts.shutdown();
        }
    }
}

File: app/src/main/java/ai/elimu/herufi/BaseApplication.kt
package ai.elimu.herufi

import android.app.Application
import android.speech.tts.TextToSpeech
import android.speech.tts.TextToSpeech.OnInitListener
import android.util.Log

class BaseApplication : Application() {
    var tts: TextToSpeech? = null

    override fun onCreate() {
        Log.i(javaClass.name, "onCreate")
        super.onCreate()

        // Initialize Text-to-Speech
        tTS
    }

    val tTS: TextToSpeech
        get() {
            if (tts == null) {
                tts = TextToSpeech(applicationContext, object : OnInitListener {
                    override fun onInit(status: Int) {
                        Log.i(javaClass.name, "TextToSpeech onInit")
                    }
                })
            }
            return tts as TextToSpeech
        }

    override fun onTerminate() {
        Log.i(javaClass.name, "onTerminate")
        super.onTerminate()

        // Release the resources used by the TextToSpeech engine
        if (tts != null) {
            tts!!.shutdown()
        }
    }
}
